### PR TITLE
Eq 1307 add unit pattern to library

### PR DESCRIPTION
**Pull request by shahi645**
Adding the unit patterns from survey runner in to sdc design pattern library. Updated readme for fractal docs. 

To test:
- run yarn start on this branch
- unit should be added under 'Form Elements'

### CODE REVIEW

File: components/02-form-elements/unit/unit.hbs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<div class="field">
+  {{> @label}}
+  <div class="input-type input-type--unit">
+    <input data-qa="input-number" id="{{id}}" class="input input--number input-type__input" name="{{id}}" value="" pattern="[0-9]*">
+    <span class="input-type__type" id="{{id}}-type">{{unit}}</span>
```
Comment (armstrongb) [inline components/02-form-elements/unit/unit.hbs]: I would like to amend `<span class="input-type__type" ...` to an `abbr` element. 
`<abbr title="{{label}}" class="input-type__type" id="{{id}}-type">{{unit}}</abbr>`

Comment (armstrongb) [reply components/02-form-elements/unit/unit.hbs]: The `<abbr>` inside an input-type will require the `text-decoration` css attribute set to none.

**Comment by armstrongb** [comment]
@shahi645 there is an example of the documentation pattern in the `panel` component.

**Comment by collisdigital** [comment]
Ensure this (and other PRs) are squashed before merging. @armstrongb do you know how to do this?

**Comment by armstrongb** [comment]
@collisdigital nope so some pointers would be good.

**Comment by shahi645** [comment]
blocked by https://github.com/ONSdigital/sdc-global-design-patterns/pull/29

File: components/02-form-elements/unit/unit.config.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    "name": "square-kilometres",
+    "context": {
+      "label": "Square-kilometres",
+      "unit": " km²"
+    },
+  }, {
+    "name": "square-miles",
+    "context": {
+      "label": "Square-miles",
+      "unit": " sq mi"
```
Comment (bennpearson) [inline components/02-form-elements/unit/unit.config.js]: Could use `mi²` here to help it fit into box

Comment (shahi645) [reply components/02-form-elements/unit/unit.config.js]: https://github.com/ONSdigital/sdc-global-design-patterns/pull/29 will amake the box bigger, as we did in survey runner.

**Pull request merged**